public/app/core/components/Login/LoginCtrl.tsx

```diff
@@ -104,9 +104,17 @@ export class LoginCtrl extends PureComponent<Props, State> {
     const params = this.props.routeParams;
     // Use window.location.href to force page reload
     if (params.redirect && params.redirect[0] === '/') {
-      window.location.href = config.appSubUrl + params.redirect;
+      if (config.appSubUrl !== '' && !params.redirect.startsWith(config.appSubUrl)) {
```

Comment:
Do we need frontend logic at all for redirect, couldn't the backend make sure to always return the correct url? Wait why are we using redirect query param here - thought we always had redirect in cookie?